sklearn/metrics/tests/test_score_objects.py

```diff
@@ -306,46 +308,87 @@ def test_make_scorer():
         make_scorer(f, needs_threshold=True, needs_proba=True)
 
 
-def test_classification_scores():
-    # Test classification scorers.
+@pytest.mark.parametrize('scorer_name, metric', [
+    ('f1', f1_score),
+    ('f1_weighted', partial(f1_score, average='weighted')),
+    ('f1_macro', partial(f1_score, average='macro')),
+    ('f1_micro', partial(f1_score, average='micro')),
+    ('precision', precision_score),
+    ('precision_weighted', partial(precision_score, average='weighted')),
+    ('precision_macro', partial(precision_score, average='macro')),
+    ('precision_micro', partial(precision_score, average='micro')),
+    ('recall', recall_score),
+    ('recall_weighted', partial(recall_score, average='weighted')),
+    ('recall_macro', partial(recall_score, average='macro')),
+    ('recall_micro', partial(recall_score, average='micro')),
+    ('jaccard', jaccard_score),
+    ('jaccard_weighted', partial(jaccard_score, average='weighted')),
+    ('jaccard_macro', partial(jaccard_score, average='macro')),
+    ('jaccard_micro', partial(jaccard_score, average='micro')),
+    ('top_k_accuracy', top_k_accuracy_score),
+])
+def test_classification_binary_scores(scorer_name, metric):
```

Comment:
The downsize of using `parametrize` here is that `clf` is trained repeatedly over the same dataset. But since the dataset is only 100 samples, I think this is okay.